components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/client/AbstractClientBase.java

```diff
@@ -87,6 +90,10 @@ public void start() throws Exception {
 
     @Override
     public void stop() throws Exception {
+        if (!inflightRequests.isTerminated()) {
+            inflightRequests.arriveAndAwaitAdvance();
+        }
+
```

Comment:
Ah yeah I would guess so if for some reason there is no callback then wont it block there?